pkg/kubelet/cm/topologymanager/policy.go

```diff
@@ -127,7 +129,12 @@ func maxOfMinAffinityCounts(filteredHints [][]TopologyHint) int {
 	return maxOfMinCount
 }
 
-func compareHints(bestNonPreferredAffinityCount int, current *TopologyHint, candidate *TopologyHint) *TopologyHint {
+func compareHints(bestNonPreferredAffinityCount int, current *TopologyHint, candidate *TopologyHint, isCandidateFitter fitterHintFunc) *TopologyHint {
+	// default isCandidateHintFitter to take narrower bitmask
+	if isCandidateFitter == nil {
+		isCandidateFitter = defaultIsCandidateFitterFunc
+	}
+
```

Comment:
Can this not be done outside of this function such that it is never nil?

Reply:
Yes, will change the code.